src/core/dispatcher/Dispatcher.ts

```diff
@@ -1,48 +1,147 @@
+     */
+    register(handlers: PluginHandlers, commands: Commands): void {
+        Object.keys(handlers).forEach((handlerType: string): void => {
+            Object.keys(handlers[handlerType]).forEach((actionName: string): void => {
+                const type = 'intent'; // todo: use handlerType (right now only one)
+                const id: ActionIdentifier = type + '.' + actionName;
+                const commandIdentifier: CommandIdentifier = handlers.intents[actionName];
+                const command = commands[commandIdentifier];
+                if (command) {
+                    this._register(id, command);
+                } else {
+                    const info = id.split('.');
+                    console.warn(
+                        `Cannot register ${info[0]} "${info[1]}" with the ` +
+                            `command "${commandIdentifier}": that command ` +
+                            `does not exist or was not registered.`,
+                    );
+                }
+            });
+        });
     }
-
     // Removes a callback based on its token.
     unregister(token: HandlerToken): void {
-        delete this.handlers[token];
+        // TODO
+        token;
+        // delete this.handlers[token];
     }
-
     // Waits for the callbacks specified to be invoked before continuing
     // execution of the current callback. This method should only be used by a
     // callback in response to a dispatched payload.
     waitFor(tokens: HandlerToken[]): Promise<HandlerToken[]> {
         // TODO
         return Promise.resolve(tokens);
     }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return the `ActionHandler` corresponding to the given `handlerToken` in
+     * the registry.
+     *
+     * @param {HandlerToken} handlerToken
+     * @returns {ActionHandler}
+     */
+    _getHandler(handlerToken: HandlerToken): ActionHandler {
+        const recordID: ActionIdentifier = Object.keys(this.registry).find(
+            (id: ActionIdentifier): boolean => {
+                const handlers = this.registry[id];
+                return !!handlers[handlerToken];
+            },
+        );
+        return this.registry[recordID][handlerToken];
+    }
+    /**
+     * Return a copy of an action's `Handlers`. Also looks into '*' records for
+     * the given action type, unless `onlyIncludeSpecific` is true.
+     * '*' is a magic key used to listen to action of a given type.
+     *
+     * @param {ActionIdentifier} actionIdentifier
+     * @param {boolean} [onlyIncludeSpecific] if true, do not include '*' records
+     * @returns {Handlers}
+     */
+    _getHandlers(actionIdentifier: ActionIdentifier, onlyIncludeSpecific = false): Handlers {
+        const record = Object.assign({}, this.registry[actionIdentifier]) || {};
+        if (!onlyIncludeSpecific) {
```

Comment:
I figured we should have a way to retrieve only the handlers that are specific to the given action identifier, simply because it's not exactly intuitive that `_getHandlers` would actually include wildcard handlers (even though in most cases that is in fact what we want). You're absolutely right about the name though.